src/states/BeforePreroll.js

```diff
@@ -30,7 +33,10 @@ export default class BeforePreroll extends ContentState {
     player.ads.debug('Received play event (BeforePreroll)');
 
     // Don't start content playback yet
-    cancelContentPlay(player);
+    if (!isMiddlewareMediatorSupported()) {
```

Comment:
Maybe you already planned on doing this, but the next step is to move this logic into cancelContentPlay.js. That is, always `cancelContentPlay` on `play` if `!isMiddlewareMediatorSupported`. Then all the states have to do is set `_shouldBlockPlay`.